From: Johannes Berg <johannes.berg@intel.com>
Date: Tue, 3 Mar 2026 15:16:15 +0100
Subject: [PATCH] wifi: nl80211: fix UHR capability validation
References: bsc#1258210
Patch-mainline: 6.18.52
Git-commit: a140826caa2c14aa5a9a6990e514c5edbdb7eafd

[ Upstream commit a140826caa2c14aa5a9a6990e514c5edbdb7eafd ]

The ieee80211_uhr_capa_size_ok() function returns a boolean,
but we need an error code here. Fix that.

Fixes: 072e6f7f416f ("wifi: cfg80211: add initial UHR support")
Cc: <stable+noautosel@kernel.org> # no drivers with UHR yet
Link: https://patch.msgid.link/20260303151614.e87ea9995be5.Ie164040a51855a3e548f05f0d0291d7d7993c7ee@changeid
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Robert Frohl <rfrohl@suse.com>
---
 net/wireless/nl80211.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 9c713094f0a5..232218818ed7 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -372,7 +372,9 @@ static int validate_uhr_capa(const struct nlattr *attr,
 	const u8 *data = nla_data(attr);
 	unsigned int len = nla_len(attr);
 
-	return ieee80211_uhr_capa_size_ok(data, len, false);
+	if (!ieee80211_uhr_capa_size_ok(data, len, false))
+		return -EINVAL;
+	return 0;
 }
 
 static int validate_uhr_operation(const struct nlattr *attr,
-- 
2.53.0

